packages/@cdktf/hcl2cdk/lib/types.ts

```diff
@@ -19,6 +19,7 @@ export type ProgramScope = {
   >;
   // Temporary flag to indicate if we need to import the cdktf library to access the token class
   hasTokenBasedTypeCoercion: boolean;
+  nodeIds: string[]; // temporarily added until replaced
```

Comment:
We only need nodeIds because it made regex based parsing easier (knowing what you are looking for). We now have a proper expression parser so we need to replace the function searching for references with it.